test/fixtures/terraform-backend-file/terraform.backend

```diff
@@ -0,0 +1,4 @@
+terraform {
+  backend "local" {}
+}
```

Comment:
Reading the docs, I think this is supposed to be the contents of the `config` block. See the comment about backend files in https://www.terraform.io/docs/backends/config.html#partial-configuration, specifically

> A backend configuration file has the contents of the backend block as top-level attributes, without the need to wrap it in another terraform or backend block:

Reply:
Ahhh yes you can only configure the backend specified in the module using `--backend-config`. You can't dynamically pick and choose the actual backend. To do that, you will need something like [terragrunt](https://terragrunt.gruntwork.io/) which can generate the remote state config dynamically before `terraform` is called. You can do this with a `generate` block that adjusts content based on an environment variable (controlled with `get_env`).

Note that `terratest` supports calling `terragrunt` by setting the `TerraformBinary` field in `terraform.Options`.

---

I still think this specific PR would be a useful addition to have.